projects/igniteui-angular/src/lib/grid/grid.component.spec.ts

```diff
@@ -1,707 +1,871 @@
 
-        expect(grid.columns[0].width).not.toBeLessThan(136);
-        expect(grid.columns[2].width).not.toBeLessThan(136);
-        expect(grid.width).toMatch('100%');
-        expect(grid.rowList.length).toBeGreaterThan(0);
-    });
+            expect(grid.id).toContain('igx-grid-');
+            expect(domGrid.id).toContain('igx-grid-');
 
-    it('col width should be >=136px - grid 10x30', () => {
-        const fix = TestBed.createComponent(IgGridTest10x30Component);
-        fix.detectChanges();
+            grid.id = 'customGridId';
+            fix.detectChanges();
 
-        const grid = fix.componentInstance.gridMinDefaultColWidth;
+            expect(grid.id).toBe('customGridId');
+            expect(domGrid.id).toBe('customGridId');
+        });
 
-        expect(grid.columns[0].width).not.toBeLessThan(136);
-        expect(grid.columns[4].width).not.toBeLessThan(136);
-        expect(grid.columns[6].width).not.toBeLessThan(136);
-        expect(grid.width).toMatch('100%');
-        expect(grid.rowList.length).toBeGreaterThan(0);
-    });
+        it('should initialize a grid with autogenerated columns', () => {
+            const fix = TestBed.createComponent(IgxGridTestComponent);
+            fix.componentInstance.data = [
+                { Number: 1, String: '1', Boolean: true, Date: new Date(Date.now()) }
+            ];
+            fix.componentInstance.columns = [];
+            fix.componentInstance.autoGenerate = true;
+            fix.detectChanges();
 
-    it('col width should be >=136px - grid 30x1000', () => {
-        const fix = TestBed.createComponent(IgGridTest30x1000Component);
-        fix.detectChanges();
+            const grid = fix.componentInstance.grid;
+
+            expect(grid).toBeDefined('Grid initializing through autoGenerate failed');
+            expect(grid.columnList.length).toEqual(4, 'Invalid number of columns initialized');
+            expect(grid.rowList.length).toEqual(1, 'Invalid number of rows initialized');
+            expect(grid.columnList.first.dataType).toEqual(DataType.Number, 'Invalid dataType set on column');
+            expect(grid.columnList.find((col) => col.index === 1).dataType)
+                    .toEqual(DataType.String, 'Invalid dataType set on column');
+            expect(grid.columnList.find((col) => col.index === 2).dataType)
+                    .toEqual(DataType.Boolean, 'Invalid dataType set on column');
+            expect(grid.columnList.last.dataType).toEqual(DataType.Date, 'Invalid dataType set on column');
+        });
 
-        const grid = fix.componentInstance.gridMinDefaultColWidth;
-        expect(grid.columns[0].width).not.toBeLessThan(136);
-        expect(grid.columns[4].width).not.toBeLessThan(136);
-        expect(grid.columns[14].width).not.toBeLessThan(136);
-        expect(fix.componentInstance.isHorizonatScrollbarVisible()).toBe(true);
-    });
+        it('should initialize a grid and emit event on column creation', () => {
```

Comment:
@ChronosSF You can move this test asserts in the "should initialize a grid with autogenerated columns" and "should initialize a grid with columns from markup" tests/